Add Legend-State to local-first.mdx (#31536)
expo/expo: docs/pages/guides/local-first.mdx

One way to think of local-first tools is to group them by the following categories: persistence, state management, and syncing. Some tools will fit into multiple categories if they handle multiple aspects of the problem. Syncing can be further subdivided into syncable data structures and transport layers.

### Legend-State

[Legend-State](https://legendapp.com/open-source/state/v3/) is a super fast all-in-one state and sync library that lets you write less code to make faster apps. Legend-State has four primary goals:

> - The fastest React state library.
> - Fine-grained reactivity for minimal renders.
> - Powerful sync and persistence (with Supabase support built-in)

Reviewer: ```suggestion
- The fastest React state library
- Fine-grained reactivity for minimal renders
- Powerful sync and persistence (with Supabase support built-in)
```

> It works with Expo and React Native (via [`react-native-mmkv`](https://github.com/mrousavy/react-native-mmkv)). This makes it a perfect match for building local-first mobile and web apps. Get started by using the [Legend-State Supabase example](https://github.com/expo/examples/tree/master/with-legend-state-supabase): `npx create-expo-app --example with-legend-state-supabase`.

Author: ```suggestion
It works with Expo and React Native (via [React Native Async Storage](https://github.com/react-native-async-storage/async-storage?tab=readme-ov-file#react-native-async-storage)). This makes it a perfect match for building local-first mobile and web apps. Get started by using the [Legend-State Supabase example](https://github.com/expo/examples/tree/master/with-legend-state-supabase): `npx create-expo-app --example with-legend-state-supabase`.
```

One way to think of local-first tools is to group them by the following categories: persistence, state management, and syncing. Some tools will fit into multiple categories if they handle multiple aspects of the problem. Syncing can be further subdivided into syncable data structures and transport layers.

> ### Legend-State

Reviewer: To solve the warning here, we'll add this to the exception in our vale rules. I'll follow up with a fix for this in a separate PR.

[Legend-State](https://legendapp.com/open-source/state/v3/) is a super fast all-in-one state and sync library that lets you write less code to make faster apps. Legend-State has four primary goals:

> 1. As easy as possible to use.
> 1. The fastest React state library.
> 1. Fine-grained reactivity for minimal renders.
> 1. Powerful sync and persistence (with Supabase support built in!)

Reviewer: ```suggestion
- The fastest React state library.
- Fine-grained reactivity for minimal renders.
- Powerful sync and persistence (with Supabase support built-in)
```

We try to reserve numbered steps when describing a procedure. I think first point is a bit opinionated on whether someone finds a particular library easy to use, in general. Also don't think it adds anything relevant here since you've already mentioned that "lets you write less code to make faster apps" in the intro.

Let's also avoid using `!` since using brackets to convey the info about the built-in support has enough emphasis.

> It works with Expo React Native (via [`react-native-mmkv`](https://github.com/mrousavy/react-native-mmkv)) out of the box! This makes it a perfect match for building local-first web and mobile apps! Get started by using the [Legend-State Supabase example](https://github.com/expo/examples/tree/master/with-legend-state-supabase): `npx create-expo-app --example with-legend-state-supabase`.

Reviewer: ```suggestion
It works with Expo and React Native (via [`react-native-mmkv`](https://github.com/mrousavy/react-native-mmkv)) out of the box. This makes it a perfect match for building local-first mobile and web apps! Get started by using the [Legend-State Supabase example](https://github.com/expo/examples/tree/master/with-legend-state-supabase): `npx create-expo-app --example with-legend-state-supabase`.
```
